sdk/python/lib/pulumi/resource.py

```diff
@@ -694,11 +703,12 @@ def __init__(self,
             if not custom:
                 raise Exception(
                     "Cannot read an existing resource unless it has a custom provider")
-            read_resource(self, t, name, props, opts)
+            res = cast('CustomResource', self)
```

Comment:
aside: (and very much a nit) my preference would be to inline these if possible.  i.e. `read_resource(cast('CustomResource', self), t, name, props, opts)`

The less locals there are, the easier it is for me to track the data flow and implications of this sort of hting.